cli/repl.c

```diff
@@ -29,6 +30,19 @@ const char* read_line(uint32_t* length) {
   return mem;
 }
 
+// Reads a line one character at a time and stop reading when we hit the
+// EOF or a new line. The buffer resizes itself when count == capacity.
```

Comment:
Everything looks good, just a little suggestion on the comment.
> The buffer resizes itself when count == capacity

This is unnecessary because how ByteBuffer works is a hidden implementation detail, "the abstraction" that the user (here this function) of a byte buffer shouldn't know about.

Reply:
Okay, will remove that. Thanks!